File: case_studies/goes/tsplot.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.backends.backend_pdf import PdfPages
from read import get_goes

pdf = PdfPages("tsplot_py.pdf")

for year in [2017, 2019]:

    df = get_goes(year)
    df["Time"] = pd.to_datetime(df[["Year", "Month", "Day"]])
    df["DayofYear"] = [x.dayofyear for x in df.Time]
    dx = df.groupby("DayofYear").agg({"Flux1": [np.min, np.max], "Flux2": [np.min, np.max]})

    for vn in ["Flux1", "Flux2"]:
        plt.clf()
        plt.axes([0.12, 0.12, 0.75, 0.8])
        plt.grid(True)
        plt.plot(dx.index, np.log10(dx[(vn, "amin")]), "-", alpha=0.5)
        plt.plot(dx.index, np.log10(dx[(vn, "amax")]), "-", alpha=0.5)
        plt.axhline(-6, label="C", color="green")
        plt.axhline(-5, label="M", color="orange")
        plt.axhline(-4, label="X", color="red")
        ha, lb = plt.gca().get_legend_handles_labels()
        leg = plt.figlegend(ha, lb, "center right")
        leg.draw_frame(False)
        plt.xlabel("Day of year", size=15)
        plt.ylabel(vn, size=15)
        plt.title(year)
        pdf.savefig()

pdf.close()

File: case_studies/goes/tsplot.R
library(dplyr)
library(ggplot2)
library(tidyr)
library(lubridate)
source("read.R")

pdf("tsplot_r.pdf")

for (year in c(2017, 2019)) {
    df = get_goes(year)
    df$Time = make_datetime(df$Year, df$Month, df$Day)
    df$DayOfYear = yday(df$Time)

    for (vn in c("Flux1", "Flux2")) {
        df$lFlux = log10(df[[vn]])
        dfs = df %>% group_by(DayOfYear) %>% summarize(maxFlux = max(lFlux), minFlux=min(lFlux))
        dfs = gather(dfs, mtype, flux, minFlux:maxFlux)
        plt = ggplot(aes(x=DayOfYear, y=flux, by=mtype, color=mtype), data=dfs) + geom_line()
        plt = plt + geom_hline(yintercept=-6, color="green")
        plt = plt + geom_hline(yintercept=-5, color="orange")
        plt = plt + geom_hline(yintercept=-4, color="red")
        plt = plt + ggtitle(sprintf("%4d %s", year, vn))
        print(plt)
    }
}

dev.off()
